lib/kitchen/provider/machine.rb

```diff
@@ -0,0 +1,155 @@
+# -*- encoding: utf-8 -*-
+#
+# Author:: Salim Afiune (<salim@afiunemaya.com.mx>)
+#
+# Copyright (C) 2014, Salim Afiune
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'mixlib/shellout'
+
+module Kitchen
+
+  module Provider
+
+    module VirtualBox
+      # This class will simulate one Virtual Box Machine
+      # It will give us the latest Forwarded Ports list
+      # where we can check if the port has changed
+      #
+      # Supported Virtual Box
+      #   => version 4.2.x
+      #   => version 4.3.x
+      class Machine
+        # The UUID of the virtual machine
+        attr_reader :uuid
+
+        # The forwarded_ports of the virtual machine
+        attr_reader :forwarded_ports
+
+
+        def initialize(root_path, uuid=nil)
+          # Did we get the UUID
+          @uuid = uuid
+
+          # The VBoxID that Vagrant spun up
+          id = File.expand_path(File.join(root_path,
+            %w{.vagrant machines default virtualbox id}))
+
+          # If UUID was not given but the VBoxID exist.
+          @uuid = File.read(id) if File.exists?(id) && @uuid.nil?
+
+          # Report that the VM is missing.
+          vm_notfound unless exists?
+
+          yield self if block_given?
+        end
+
+        def execute(*command)
+          # Easy shellout method to execute something! :D
+          command.insert(0, vboxmanage)
+          c = Mixlib::ShellOut.new(*command.join(' '), :timeout => 10800)
```

Comment:
Dunno why that doesn't work. But I have added some tests to validate functionality. :smile: